docs/how-xtrabackup-works.md

```diff
@@ -1,123 +1,149 @@
+
+The `BACKUP_ADMIN` privilege is required to execute the `LOCK INSTANCE FOR BACKUP` statement. This privilege manages database backups, ensures data consistency, and performs backup-related tasks.
+
+### Locking the binary log
+
+The `performance_schema.log_status` table in MySQL is a valuable resource for backups. This table provides critical information that 
+allows a backup tool to copy necessary log files without locking the resources for the duration of the copy process. When queried, 
+the `log_status` table causes the server to momentarily block logging and related administrative changes to populate the table 
+with data, and then the server releases the resources. This table identifies the point in the 
+source's binary log and `gtid_executed` record the backup should copy, including the relay log for each replication channel. 
+Additionally, it offers essential details, such as the last log sequence number (LSN) and the 
+LSN of the previous checkpoint for InnoDB.
+
+To use the `log_status` table effectively, you must have the `BACKUP_ADMIN` privilege and the `SELECT` privilege.
+
+The table contains several columns, including `SERVER_UUID`, which is the server's unique identifier, and `LOCAL`, which provides the log 
+position state information from the source as a JSON object. This table includes the current binary log file name and position, 
+as well as the `gtid_executed` value at the time the table was accessed. The `REPLICATION` section is a JSON array containing detailed information about each replication channel, such as the channel name, current relay log file, and position. The `STORAGE_ENGINES` section provides 
+JSON objects with keys for each applicable storage engine, detailing their specific log information.
+
+To perform a backup using the `log_status` table, one would typically follow these steps:
+1. Ensure you have the necessary privileges to access the `log_status` table.
+2. Query the `log_status` table to retrieve the current logging state.
+3. Use the information from the `LOCAL` and `REPLICATION` sections to determine the log files and positions that need to be copied.
+4. Copy the log files up to the specified positions without the need to lock the resources for an extended period.
+5. Use the information from the `STORAGE_ENGINES` section to handle engine-specific log data, if applicable.
+
+It's important to note that the `log_status` table does not have indexes, and you cannot do operations like `TRUNCATE TABLE`.
+
+
+## Prepare phase
+
+The `prepare` phase uses the MySQL crash recovery process. This automated process uses transaction logs to return the database to a consistent state. The process works in the following way:
+
+* MySQL checks the InnoDB transaction log files to determine the database state
+
+* MySQL applies any uncommitted changes from the transaction logs to the database
+
+Percona XtraBackup performs crash recovery in the prepare phase before restoring the database. 
```

Comment:
this is very generic imho. my blog post (or our docs somewhere gives better info).

Essentially it si about applying redo log for InnoDb datafiles and then applying undo logs to do the rollback of uncommitted transactions.

For MyISAM tables, there is nothing to be done in prepare phase as we ensured that we copied under proper lock during backup phase.